lib/nostrum/shard/session.ex

```diff
@@ -61,7 +61,7 @@ defmodule Nostrum.Shard.Session do
     {:ok, worker} = :gun.open(:binary.bin_to_list(gateway), 443, @gun_opts)
     {:ok, :http} = :gun.await_up(worker, @timeout_connect)
     stream = :gun.ws_upgrade(worker, @gateway_qs)
-    await_ws_upgrade(worker, stream)
+    {:upgrade, ["websocket"], _} = :gun.await(worker, stream, @timeout_ws_upgrade)
```

Comment:
Are there alternative return values for this function now we should handle? Previously we had a whole section of code that provided detailed errors when things went awry.

Reply:
Only `:gun_error` and a timeout, which will cause an exit with badmatch.